mount/index.ts

```diff
@@ -33,69 +32,143 @@ export function mountLazy <T extends mojave.Mountable>(selector: string, importe
-                    parent.removeChild(node);
-                }
-
-                // render
-                render(
-                    createElement(mountable as ComponentType<any>, params),
-                    parent,
-                    opts.hydrate ? node : undefined
-                );
-            }
-            else if ("class" === options.type)
-            {
-                let standaloneOptions = options as mojave.ClassMountOptions;
-                const mounted = new mountableAny(node, ...(standaloneOptions.params || []));
-                mounted.init();
-            }
-            else
-            {
-                let standaloneOptions = options as mojave.FunctionMountOptions;
-                (mountable as Function)(node, ...(standaloneOptions.params || []));
-            }
-        }
+    let elements = find(selector);
+
+    if (!elements.length)
+    {
+        return;
+    }
+
+    importer().then(
+        module => {
+            elements.forEach(element => {
+                doMountClass<T>(element, module.default, options);
+            });
+        },
+        error => console.error(`Mounting of component of path '${selector}' failed: ${error.message}`, error)
     );
 }
 
 
+/**
+ * Mounts a StandaloneComponent
+ */
+function doMountClass<T extends mojave.MountableClass>(node: HTMLElement, mountable: T, options?: mojave.ClassMountOptions<T>): void
+{
+    options = options || {};
+    const mounted = new mountable(node, ...(options.params || []));
+    mounted.init();
+}
+
+
+/**
+ * Mounts a function into all elements matching the given selector.
+ */
+export function mount<T extends mojave.MountableFunction>(selector: string, mountable: T, options?: mojave.FunctionMountOptions<T>): void
+{
+    find(selector).forEach(node => {
+        doMountFunction<T>(node, mountable, options);
+    });
+}
+
+
+/**
+ * Mounts the component lazily, if an element matching the selector exists.
+ *
+ * The importer must import the component.
+ * Example:
+ *
+ *     mountLazy(".selector", () => import("./src/MyComp"));
+ */
+export function mountLazy <T extends mojave.MountableFunction>(selector: string, importer: () => Promise<any>, options?: mojave.FunctionMountOptions<T>) : void
+{
+    let elements = find(selector);
+
+    if (!elements.length)
+    {
+        return;
+    }
+
+    importer().then(
+        module => {
+            elements.forEach(element => {
+                doMountFunction<T>(element, module.default, options);
+            });
+        },
+        error => console.error(`Mounting of component of path '${selector}' failed: ${error.message}`, error)
+    );
+}
+
 
+/**
+ * Mounts a function
+ */
+function doMountFunction<T extends mojave.MountableFunction>(node: HTMLElement, mountable: T, options?: mojave.FunctionMountOptions<T>): void
+{
+    options = options || {};
+    mountable(node, ...(options.params || []));
```

Comment:
Can be inlined

Reply:
We talked about this in private: All `doMount*` functions are used in their corresponding eager and lazy versions, which is why we keep them as they are.